### PR TITLE
CASMHMS-5911

### PR DESCRIPTION
**Pull request by rsjostrand-hpe**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_
Added the field `hmcollector_external_hostname` to values.yaml to specify a hostname to address the collector by. For example this would be populated with `hmcollector.hmnlb.mug.dev.cray.com`. The collector will still continue to use IP address when it sets up subscriptions, but it will also be accessible at `hmcollector.hmnlb.$SYSTEM_NAME.$SITE_DOMAIN`.

This commit contains the changes from the latest 2.15 chart and 2.16.0: https://github.com/Cray-HPE/hms-hmcollector-charts/pull/20/commits/07a60eee56355c60b438dded7a857d79eb3206b5

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_
backwards compatible

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Partially resolves CASMHMS-5911

## Testing

_List the environments in which these changes were tested._

Tested on:
* `Mug`

Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?

### Before
Here is the existing virtual service for the collector.
```
ncn-m001:~ #  kubectl -n services get vs cray-hms-hmcollector-ingress
NAME                           GATEWAYS          HOSTS              AGE
cray-hms-hmcollector-ingress   ["hmn-gateway"]   ["10.94.100.71"]   138d
```

Here is the behavior of the of the IP address `10.94.100.71`. The `JSON payload malformed!` message means that the collector was reached. 
```bash
ncn-m001:~ # curl -X POST -d {} http://10.94.100.71 -i
HTTP/1.1 400 Bad Request
date: Tue, 31 Jan 2023 20:44:46 GMT
content-length: 23
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 1
server: istio-envoy

JSON payload malformed!

ncn-m001:~ # curl -X POST -d {} -k  https://10.94.100.71 -i
HTTP/2 400
date: Tue, 31 Jan 2023 20:45:00 GMT
content-length: 23
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 9
server: istio-envoy

JSON payload malformed!

```

Here is the behavior of `hmcollector.hmnlb.mug.dev.cray.com`. The 404 is coming from istio. 
```
ncn-m001:~ # curl -X POST -d {} http://hmcollector.hmnlb.mug.dev.cray.com -i
HTTP/1.1 404 Not Found
date: Tue, 31 Jan 2023 20:44:36 GMT
server: istio-envoy
connection: close
content-length: 0

ncn-m001:~ # curl -X POST -d {} https://hmcollector.hmnlb.mug.dev.cray.com -i
HTTP/2 404
date: Tue, 31 Jan 2023 20:44:41 GMT
server: istio-envoy
```

### After
Here is the existing virtual service for the collector with the system hostname.
```
ncn-m001:~ # kubectl -n services get vs cray-hms-hmcollector-ingress
NAME                           GATEWAYS          HOSTS                                                   AGE
cray-hms-hmcollector-ingress   ["hmn-gateway"]   ["10.94.100.71","hmcollector.hmnlb.mug.dev.cray.com"]   138d
```

Here is the behavior of the of the IP address `10.94.100.71`. The `JSON payload malformed!` message means that the collector was reached. 

```
ncn-m001:~ # curl -X POST -d {} http://10.94.100.71 -i
HTTP/1.1 400 Bad Request
date: Tue, 31 Jan 2023 20:59:28 GMT
content-length: 23
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 0
server: istio-envoy

JSON payload malformed!

ncn-m001:~ # curl -X POST -d {} -k  https://10.94.100.71 -i
HTTP/2 400
date: Tue, 31 Jan 2023 20:59:29 GMT
content-length: 23
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 1
server: istio-envoy

JSON payload malformed!
```

Here is the behavior of `hmcollector.hmnlb.mug.dev.cray.com`. The `JSON payload malformed!` message means that the collector was reached. 

```
ncn-m001:~ # curl -X POST -d {} http://hmcollector.hmnlb.mug.dev.cray.com -i
HTTP/1.1 400 Bad Request
date: Tue, 31 Jan 2023 20:59:28 GMT
content-length: 23
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 1
server: istio-envoy

JSON payload malformed!

ncn-m001:~ # curl -X POST -d {} https://hmcollector.hmnlb.mug.dev.cray.com -i
HTTP/2 400
date: Tue, 31 Jan 2023 20:59:28 GMT
content-length: 23
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 1
server: istio-envoy

JSON payload malformed!
```

### As an event destination
I created an event subscription on a BMC and had it pointed toward the collector at `hmcollector.hmnlb.mug.dev.cray.com`.

```
ncn-m001:~ # curl -u root:initial0 -k -i -X POST -d '{
  "Context": "x3000c0s19b4",
  "Destination": "http://hmcollector.hmnlb.mug.dev.cray.com/x3000c0s19b4",
  "EventTypes": [
    "Alert"
  ],
  "Protocol": "Redfish"
}' https://x3000c0s19b4/redfish/v1/EventService/Subscriptions


HTTP/1.1 200 OK
Content-Type: application/json
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-UA-Compatible: IE=11
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
Cache-Control: no-store, no-cache
Pragma: no-cache
Content-Security-Policy: default-src 'self'; object-src 'none'
Content-Length: 166
Date: Tue, 31 Jan 2023 22:29:58 GMT

{"@odata.type":"#Message.v1_0_5.Message","MessageId":"Base.1.1.0.Created","Message":"The resource has been created successfully","Severity":"OK","Resolution":"None"}
```

Verified that power on/off events made it onto the kafka bus:
```
{"@odata.context":"/redfish/v1/$metadata#Event.Event","@odata.id":"/redfish/v1/EventService/Events","@odata.type":"#Event.v1_2_1.Event","Id":"Event1","Name":"Event Array","Context":"x3000c0s19b4","Description":"Events","Events":[{"EventType":"Alert","EventId":"Event1","EventTimestamp":"2023-01-31T22:30:57+00:00","Severity":"OK","Message":"System Power Turned On.","MessageId":"Alert.1.0.0.SystemPowerOn","MessageArgs":[""],"OriginOfCondition":{"@odata.id":"/redfish/v1/Systems/"}}]}
```

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable

### CODE REVIEW

File: charts/v2.16.0/cray-hms-hmcollector/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# by default we'll ignore any subcharts included, but simply adjust this if need be
```
Comment (shunr-hpe) [inline charts/v2.16.0/cray-hms-hmcollector/.gitignore:1]: Is charts/v2.16.0 the right directory name? I expected charts/v2.16 instead?

**Review by shunr-hpe — APPROVED** [review]

**Comment by alvarez3-hpe** [comment]
Doesn't cray-hms-hmcollector.compatibility.yaml need to get updated?

File: charts/v2.16.0/cray-hms-hmcollector/.gitignore (hunk 1 above)
Comment (rsjostrand-hpe) [reply charts/v2.16.0/cray-hms-hmcollector/.gitignore]: You are right. Moved it to the right spot.

**Comment by rsjostrand-hpe** [comment]
Updated `cray-hms-hmcollector.compatibility.yaml`

**Review by alvarez3-hpe — APPROVED** [review]

**Pull request merged**